test_conformance/math_brute_force/main.c

```diff
@@ -1745,9 +1755,18 @@ int MakeKernels( const char **c, cl_uint count, const char *name, cl_uint kernel
 {
     int error = 0;
     cl_uint i;
-    char options[200];
-
-    strcpy(options, "-cl-std=CL2.0");
+    char options[200] = "";
+    size_t major = 0;
+    size_t minor = 0;
+    error = get_device_version(gDevice, &major, &minor);
+    if (error != CL_SUCCESS)
+    {
+        vlog_error("\t\tFAILED -- Failed to get devcie version. (%d)\n", error);
+        return error;
+    }
+    if (major >= 2) {
+        strcpy(options, "-cl-std=CL2.0");
+    }
```

Comment:
A couple of comments:
- This duplicates even more code. We should implement `MakeKernels` using `MakeKernel` before making further changes.
- I couldn't spot any new built-in function added on master compared to cl12_trunk. Do we even need to pass `-cl-std=CL2.0` at all?

Reply:
I assume that each test which is build on device 2.X should add to this flag to the build options? Am I wrong ?

Reply:
Well, it's only really required for programs that use 2.x features.